### PR TITLE
Add option to generate public input column, update input file format

### LINKED ISSUE #489: [Feature request]: Update public input format
Public input JSON files should contain all necessary information for extracting correct public input column.
It is important for NilFoundation/evm-placeholder-verification#104 compatibility

### PR DESCRIPTION
**Pull request by makxenov**
Closes #489

### CODE REVIEW

File: bin/assigner/src/main.cpp [hunk 1]
```diff
@@ -696,7 +705,16 @@ int main(int argc, char *argv[]) {
...
     }
 
+    if (vm.count("dump-public-input")) {
+        processed_public_input_file_name = vm["dump-public-input"].as<std::string>();
+        input_only = true;
+    }
+
     if (vm.count("private-input")) {
+        if (input_only) {
+            std::cerr << "Atata" << std::endl;
```
Comment (akokoshn) [inline bin/assigner/src/main.cpp:715]: :D

File: bin/assigner/src/main.cpp [hunk 2]
```diff
@@ -614,6 +620,7 @@ int main(int argc, char *argv[]) {
             ("private-input,p", boost::program_options::value<std::string>(), "Private input file")
             ("assignment-table,t", boost::program_options::value<std::string>(), "Assignment table output file")
             ("circuit,c", boost::program_options::value<std::string>(), "Circuit output file")
+            ("dump-public-input,d", boost::program_options::value<std::string>(), "Processed public input output file")
```
Comment (CblPOK-git) [inline bin/assigner/src/main.cpp:623]: Maybe better to set it as generate-type, not as separate flag?

Comment (akokoshn) [reply bin/assigner/src/main.cpp:623]: we should provide file name for dump, not only mode

File: bin/assigner/src/main.cpp [hunk 3]
```diff
@@ -683,6 +692,8 @@ int main(int argc, char *argv[]) {
             gen_mode = nil::blueprint::generation_mode::ASSIGNMENTS;
         } else if (generate_type == "size_estimation") {
             gen_mode = nil::blueprint::generation_mode::SIZE_ESTIMATION;
+        } else if (generate_type == "public-input-column") {
```
Comment (akokoshn) [inline bin/assigner/src/main.cpp:695]: add possible value to description of `generate_type` parameter

File: bin/assigner/src/main.cpp [hunk 4]
```diff
@@ -707,7 +728,7 @@ int main(int argc, char *argv[]) {
     if (std::uint8_t(gen_mode & nil::blueprint::generation_mode::ASSIGNMENTS)) {
         if (vm.count("assignment-table")) {
             assignment_table_file_name = vm["assignment-table"].as<std::string>();
-        } else {
+        } else if (!public_input_only) {
```
Comment (akokoshn) [inline bin/assigner/src/main.cpp:731]: if `generation_mode` includes `ASSIGNMENT` - table must be generated.
We can:
1. allow mode `ASSIGNMENTS | PUBLIC_INPUT` and generate both files
2. `ASSIGNMENTS | PUBLIC_INPUT` - impossible mode (like now), in this case here `public_input_only` - never true

File: bin/assigner/src/main.cpp [hunk 5]
```diff
@@ -717,7 +738,7 @@ int main(int argc, char *argv[]) {
     if (std::uint8_t(gen_mode & nil::blueprint::generation_mode::CIRCUIT)) {
         if (vm.count("circuit")) {
             circuit_file_name = vm["circuit"].as<std::string>();
-        } else {
+        } else if (!public_input_only) {
```
Comment (akokoshn) [inline bin/assigner/src/main.cpp:741]: same concern like for ASSIGNMENTS mode

**Review by akokoshn — CHANGES_REQUESTED** [review]

File: bin/assigner/src/main.cpp (hunk 5 above)
Comment (makxenov) [reply bin/assigner/src/main.cpp]: fixed

**Review by aleasims — APPROVED** [review]

File: bin/assigner/src/main.cpp [hunk 6]
```diff
@@ -661,6 +668,8 @@ int main(int argc, char *argv[]) {
     std::string private_input_file_name;
     std::string assignment_table_file_name;
     std::string circuit_file_name;
+    std::string processed_public_input_file_name;
+    bool public_input_only = false;
```
Comment (akokoshn) [inline bin/assigner/src/main.cpp:672]: still need this var?

**Review by akokoshn — CHANGES_REQUESTED** [review]

File: bin/assigner/src/main.cpp (hunk 6 above)
Comment (makxenov) [reply bin/assigner/src/main.cpp]: removed, thanks

**Review by akokoshn — APPROVED** [review]

**Review by ETatuzova — APPROVED** [review]

**Pull request merged**